lib/livebook_web/plugs/auth_plug.ex

```diff
@@ -64,10 +73,18 @@ defmodule LivebookWeb.AuthPlug do
       |> redirect(to: path_with_query(conn.request_path, query_params))
       |> halt()
     else
-      redirect_to_authenticate(conn)
+      conn
+      |> put_redirect_path_in_session()
+      |> redirect_to_authenticate()
     end
   end
 
+  defp put_redirect_path_in_session(
+         %{request_path: request_path, query_params: query_params} = conn
+       ) do
+    put_session(conn, :redirect_to, path_with_query(request_path, query_params))
```

Comment:
You can see how this is implemented in Phoenix generators: https://github.com/phoenixframework/phoenix/blob/44f24b720c1832216a78fa1cfbe218372e67c4d1/priv/templates/phx.gen.auth/auth.ex#L147

TL;DR: use `current_path` and check that the method is GET.